src/electron/routing_service.ts

```diff
@@ -251,52 +291,68 @@ function installWindowsRoutingServices(): Promise<void> {
-  ];
-
-  // These Linux service files are located in a mounted folder of the AppImage, typically
-  // located at /tmp/.mount_Outlinxxxxxx/resources/. These files can only be acceeded by
-  // the user who launched Outline.AppImage, so even root cannot access the files or folders.
-  // Therefore we have to copy these files to a normal temporary folder, and execute them
-  // as root.
-  //
-  // Also, we are copying individual files instead of the entire folder because they are in
-  // electron's "asar" archive (which is returned by getAppPath). Electron tries to inject
-  // some logic to node's fs API and make sure the caller can access files in the archive.
-  // However directories are not supported.
-  //
-  // References:
-  // - https://github.com/AppImage/AppImageKit/issues/146
-  // - https://xwartz.gitbooks.io/electron-gitbook/content/en/tutorial/application-packaging.html
-  const tmp = await fsextra.mkdtemp('/tmp/');
-  const srcFolderPath = path.join(getAppPath(), OUTLINE_PROXY_CONTROLLER_PATH);
-
-  console.log(`copying service installation files to ${tmp}`);
-  for (const [filename, executable] of LINUX_SERVICE_FILE_NAMES) {
-    const dest = path.join(tmp, filename);
-    await fsextra.copy(path.join(srcFolderPath, filename), dest, {overwrite: true});
-    if (executable) {
-      await fsextra.chmod(dest, 0o755);
-    }
+/**
+ * The actual steps to install a Linux Outline service. The caller should already
+ * granted the admin permission to the process before calling this method.
+ */
+export async function doInstallLinuxRoutingServices(): Promise<void> {
+  if (!isLinux) {
+    throw new Error('please call this method in Linux');
   }
-  console.log(`all service installation files copied to ${tmp} successfully`);
 
-  await executeCommandAsRoot(path.join(tmp, LINUX_INSTALLER_FILENAME));
+  console.log('installing Outline service...');
+  const SERVICE_SOURCE_FOLDER = path.resolve(getAppPath(), 'tools/outline_proxy_controller/dist');
+  const SERVICE_BINARY = 'OutlineProxyController';
+  const SERVICE_BINARY_TARGET_FOLDER = '/usr/local/sbin';
+  const SERVICE_DEFINITION = 'outline_proxy_controller.service';
+  const SERVICE_DEFINITION_TARGET_FOLDER = '/etc/systemd/system';
+
+  const binSource = path.join(SERVICE_SOURCE_FOLDER, SERVICE_BINARY);
+  const binTarget = path.join(SERVICE_BINARY_TARGET_FOLDER, SERVICE_BINARY);
+  console.log(`copying Outline service binary from "${binSource}" to "${binTarget}"...`);
+  await fsextra.copy(binSource, binTarget, {overwrite: true});
+  await fsextra.chmod(binTarget, 0o755);
+  console.info('successfully installed Outline service binary');
+
+  const svcSource = path.join(SERVICE_SOURCE_FOLDER, SERVICE_DEFINITION);
+  const svcTarget = path.join(SERVICE_DEFINITION_TARGET_FOLDER, SERVICE_DEFINITION);
+  console.log(`copying Outline service definition from "${svcSource}" to "${svcTarget}"...`);
+  await fsextra.copy(svcSource, svcTarget, {overwrite: true});
```

Comment:
What are the permissions after the copy?

Reply:
It's `-rw-r--r--`.